src/solvers/jesse_dg/ESDG_2D.jl

```diff
@@ -0,0 +1,281 @@
+md = make_periodic(md,rd)
+
+struct UnstructuredMesh{NDIMS,Tv,Ti}
+    VXYZ::NTuple{NDIMS,Tv}
+    EToV::Matrix{Ti}
+end
+function Base.show(io::IO, mesh::UnstructuredMesh{NDIMS}) where {NDIMS}
+    @nospecialize mesh
+    println("Unstructured mesh in $NDIMS dimensions.")
+end
+
+# accumulate Q.*F into rhs
+function hadsum_ATr!(rhs,ATr,F::Fxn,u; skip_index=(i,j)->false) where {Fxn}
+    rows,cols = axes(ATr)
+    for i in cols
+        ui = u[i]
+        val_i = rhs[i]
+        for j in rows
+            if !skip_index(i,j)
+                val_i += ATr[j,i].*F(ui,u[j]) # breaks for tuples, OK for StaticArrays
+            end
+        end
+        rhs[i] = val_i # why not .= here?
+    end
+end
+
+function initial_condition(xyz,t,equations::CompressibleEulerEquations2D)
+    x,y = xyz
+    ρ = 1 + .5*exp(-25*(x^2+y^2))
+    # ρ = 1 + .5*sin(pi*x)*sin(pi*y)
+    u = 1.0
+    v = .5    
+    p = 2.
+    return prim2cons((ρ,u,v,p),equations)
+end
+
+eqn = CompressibleEulerEquations2D(1.4)
+F(orientation) = (uL,uR)->Trixi.flux_chandrashekar(uL,uR,orientation,CompressibleEulerEquations2D(1.4))
+Qrhskew,Qshskew,VhP,Ph = build_hSBP_ops(rd)
+QrhskewTr = Matrix(Qrhskew')
+QshskewTr = Matrix(Qshskew')
+
+# StructArray initialization - problem in entropy2cons
+staticzip(::Type{SVector{M}},x::NTuple{M,Array{T,N}}) where {M,T,N} = SVector{M}.(zip(x...))
+# xyz = StructArray(staticzip(SVector{2},md.xyz))
+# U = (xyz->initial_condition(xyz,0,eqn)).(xyz) # hack to retain tuple elements in U
+
+## ====== workaround for StructArrays/DiffEq.jl from https://github.com/SciML/OrdinaryDiffEq.jl/issues/1386
+function RecursiveArrayTools.recursivecopy(a::AbstractArray{T,N}) where {T<:AbstractArray,N}
+    if ArrayInterface.ismutable(a)
+      b = similar(a)
+      map!(recursivecopy,b,a)
+    else
+      ArrayInterface.restructure(a,map(recursivecopy,a))
+    end
+end
+ArrayInterface.ismutable(x::StructArray) = true
+## ====== end workaround ===========
+
+Base.real(rd::RefElemData) = Float64 # is this for DiffEq.jl?
+
+function Trixi.create_cache(mesh::UnstructuredMesh, equations, rd::RefElemData, RealT, uEltype)
+
+    @unpack VXYZ,EToV = mesh
+    md = MeshData(VXYZ...,EToV,rd)
+    md = make_periodic(md,rd)
+
+    # for flux differencing on general elements
+    Qrhskew,Qshskew,VhP,Ph = build_hSBP_ops(rd)
+    QrhskewTr = Matrix(Qrhskew')
+    QshskewTr = Matrix(Qshskew')
+
+    # tmp variables for entropy projection
+    Uq = StructArray(staticzip(SVector{4},ntuple(_->similar(md.xq),4)))
+    VUq = similar(Uq)
+    VUh = StructArray(staticzip(SVector{4},ntuple(_->similar([md.xq;md.xf]),4)))
+    Uh = similar(VUh)
+
+    cache = (;md,
+            QrhskewTr,QshskewTr,VhP,Ph,
+            Uq,VUq,VUh,Uh)
+
+    return cache
+end
+
+@inline function tmap!(f,out,x)
+    Trixi.@threaded for i = 1:length(x)
+        @inbounds out[i] = f(x[i])
+    end
+end
+
+project_and_store!(y,x) = mul!(y,Ph,x) # can't use matmul! b/c its applied to a subarray
```

Comment:
This can cause problems because of the global variable `Ph`. You could consider passing `Ph` as an argument or use some `let` trick or something similar.
Caveat: I didn't run the code (and cannot do that right now/), I'm just trying to judge from looking at it.

Reply:
Thanks! Good point, fixed this.